Add the regenerate-bundle-batch API endpoint

mprahl opened the pull request:
This allows a user to submit multiple regenerate-bundle requests as part of a single batch.

Additionally, this fixes the OpenAPI spec for regenerate-bundle and changes messaging.py to use the correct logger.

lcarva commented on iib/web/messaging.py:
```diff
@@ -164,53 +238,46 @@ def send_message_for_state_change(request, new_batch_msg=False):
+        else:
+            # If no request envelope was generated, then that means IIB is not configured to
+            # send request state change messages and the rest of the loop can be skipped
+            break
```
This looks like a minor optimization that may cause some unexpected errors in the future due to unrelated code changes.

lcarva commented on iib/web/messaging.py:
```diff
@@ -164,53 +238,46 @@ def send_message_for_state_change(request, new_batch_msg=False):
+            break
+
+    # Just use the last request's batch since the batch is the same for all of them
+    batch = request.batch
```
I find that using a variable, that has been defined in a for-loop statement, outside the for-loop to increase the difficulty in understanding and debugging the code. Consider taking the batch ID as a function parameter. You could even consider raising an error if any of the requests have a different batch ID.

mprahl replied:
I'll just use `batch = requests[0].batch` instead.

lcarva commented on iib/web/messaging.py:
```diff
@@ -164,53 +238,46 @@ def send_message_for_state_change(request, new_batch_msg=False):
+
+    :param list requests: the requests that were created as part of the batch request
+    """
+    if not requests:
```
Missing unit test coverage here.

yashvardhannanavati commented on iib/web/api_v1.py:
```diff
@@ -338,3 +338,60 @@ def regenerate_bundle():
+    batch = Batch()
+    db.session.add(batch)
+    requests = []
+    # Iterate through all the requests and verify that they are valid before committing them and
```
`# Iterate through all the requests` => `# Iterate through all the payloads`

mprahl commented:
@yashvardhannanavati and @lcarva it occurred to me that I should send the state change messages before the tasks are scheduled to avoid a race condition where the workers could have started on the tasks and updated the state before all the state change messages are sent out.

I just moved it up in the last push.
